### PR TITLE
Basic Frameless window support

### LINKED ISSUE #67: Frameless Window
Can we have frameless window support.

I am using a modified version of [neovim-qt](https://github.com/khalidchawtany/neovim-qt/commit/9f2af46da7c4013d7cecce68db146ee8ef047bce) in which I remove the window frame to make it clean. 
It is draggable resizable  and everything.

### PR DESCRIPTION
**Pull request by PyGamer0**
This commit will add a basic frameless window option. It is invoked by:
```
neovide --frameless
```
This window is not resizeable nor movable as of now.

### CODE REVIEW

File: src/cmd_line.rs [hunk 1]
```diff
@@ -32,6 +33,7 @@ impl Default for CmdLineSettings {
             remote_tcp: None,
             multi_grid: false,
             maximized: false,
+            frameless: true,
```
Comment (Kethku) [inline src/cmd_line.rs:36]: I'm guessing we don't want frameless to default to true. Could you change this to false?

File: src/cmd_line.rs [hunk 2]
```diff
@@ -69,6 +71,11 @@ pub fn handle_command_line_arguments() {
                 .long("multiGrid")
                 .help("Enable Multigrid"),
         )
+        .arg(
+            Arg::with_name("frameless")
+            .long("frameless")
+            .help("Enables Frameless Neovide")
```
Comment (Kethku) [inline src/cmd_line.rs:77]: I think maybe a better help message would be "Removes the window frame. Note, might make the window not resizable"

File: src/cmd_line.rs [hunk 3]
```diff
@@ -124,5 +131,6 @@ pub fn handle_command_line_arguments() {
         disowned: matches.is_present("disowned"),
         wsl: matches.is_present("wsl"),
         geometry: matches.value_of("geometry").map(|i| i.to_owned()),
+        frameless: !matches.is_present("frameless"),
```
Comment (Kethku) [inline src/cmd_line.rs:134]: I'm confused by this. If you have a frameless commandline argument, does that set frameless to false?

If thats true, does frameless remove the frame?

**Review by Kethku — COMMENTED** [review]
Just a couple quick questions. Thanks so much for looking into this!

File: src/cmd_line.rs (hunk 3 above)
Comment (PyGamer0) [reply src/cmd_line.rs:134]: if `frameless` is `false`;  then the window **is frameless**; that is why I put a not there

File: src/cmd_line.rs (hunk 2 above)
Comment (PyGamer0) [reply src/cmd_line.rs:77]: ok then

File: src/cmd_line.rs (hunk 3 above)
Comment (PyGamer0) [reply src/cmd_line.rs:134]: the same reason is why the default value is `true`

Comment (Kethku) [reply src/cmd_line.rs]: That is counterintuitive to me. I expected that frameless being false would mean there is a frame. Maybe I'm confused?

**Comment by PyGamer0** [comment]
I have changed the frameless variable to default to false. @Kethku hopefully the code now makes sense.

**Comment by j4qfrost** [comment]
Looks just about ready to merge. @PyGamer0 it would be helpful to add TODOs in sections where we can add code to make the window resizable or movable. We could add a little square that auto-hides for moving the window for example.

**Comment by Kethku** [comment]
I'll just make an issue to track that work :)

**Comment by Arelav** [comment]
Just for coming from Google as me.
You need to configure env variables.
If you run Neovide GUI not from terminal (for example from Alfred)
```
launchctl setenv NEOVIDE_FRAMELESS true
launchctl setenv NEOVIDE_FRAME none
```

**Comment by bassamsdata** [comment]
@Arelav Hey, thank you so much, I was searching for a long time for this solution thanks a lot. It worked perfectly. 🙏

**Comment by bassamsdata** [comment]
@Arelav 
Sorry to mention you again, but a just small question, did the command-v (paste) works with you? I tried a lot but no luck.
I have set this `let g:neovide_input_use_logo = v:true` but seems to not work.
Thanks

**Comment by Arelav** [comment]
> @Arelav
> Sorry to mention you again, but a just small question, did the command-v (paste) works with you? I tried a lot but no luck.
> I have set this `let g:neovide_input_use_logo = v:true` but seems to not work.
> Thanks

I usually use shared clipboard  and use just `p` to paste from yank or clipboard as well. I can’t try just now.

**Comment by bassamsdata** [comment]
Ah ok. I'm now using Ctrl-R to paste but it's a bit meh. thanks for the reply

**Pull request merged**